ecs-init/cache/cache.go

```diff
@@ -61,6 +67,20 @@ func (d *Downloader) fileNotEmpty(filename string) bool {
 	return fileinfo.Size() > 0
 }
 
+// getRegion finds region from metadata and caches for the life of downloader
+func (d *Downloader) getRegion() string {
+	if d.region == "" {
```

Comment:
Reverse the conditional so we don't need this much indentation:

```go
if d.region != "" {
	return d.region
}
// do everything else
```

I don't think there's any concurrency going on here, but a note for other times you'll do this pattern: checking for something like this or mutating it normally will require a lock to prevent race conditions.